<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DockWidgetLayout</class>
 <widget class="QDockWidget" name="DockWidgetLayout">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>229</width>
    <height>243</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <bold>true</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Layout control</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <property name="font">
    <font>
     <bold>false</bold>
    </font>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QLabel" name="lblAttract">
         <property name="text">
          <string>Attract</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lblRepulse">
         <property name="text">
          <string>Repulse</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="sldBalance">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksBelow</enum>
      </property>
      <property name="tickInterval">
       <number>25</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_4">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QLabel" name="lblHandleWeight">
         <property name="text">
          <string>Bend transitions:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="dispHandleWeight">
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="sldHandleWeight">
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksBelow</enum>
      </property>
      <property name="tickInterval">
       <number>25</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_3">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QLabel" name="lblNatLength">
         <property name="text">
          <string>Natural transition length:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="dispNatLength">
         <property name="text">
          <string>0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="sldNatLength">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksBelow</enum>
      </property>
      <property name="tickInterval">
       <number>25</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_7">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>1</number>
       </property>
       <property name="bottomMargin">
        <number>1</number>
       </property>
       <item>
        <widget class="QLabel" name="lblEnableTree">
         <property name="text">
          <string>Enable tree acceleration:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkEnableTree">
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="btnStartStop">
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>sldBalance</sender>
   <signal>valueChanged(int)</signal>
   <receiver>DockWidgetLayout</receiver>
   <slot>onAttractionChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldHandleWeight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>DockWidgetLayout</receiver>
   <slot>onHandleWeightChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>125</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldNatLength</sender>
   <signal>valueChanged(int)</signal>
   <receiver>DockWidgetLayout</receiver>
   <slot>onNatLengthChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnStartStop</sender>
   <signal>clicked()</signal>
   <receiver>DockWidgetLayout</receiver>
   <slot>onStartStop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkEnableTree</sender>
   <signal>toggled(bool)</signal>
   <receiver>DockWidgetLayout</receiver>
   <slot>onTreeToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sldBalance</sender>
   <signal>valueChanged(int)</signal>
   <receiver>DockWidgetLayout</receiver>
   <slot>onRepulsionChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onAttractionChanged(int)</slot>
  <slot>onRepulsionChanged(int)</slot>
  <slot>onNatLengthChanged(int)</slot>
  <slot>onHandleWeightChanged(int)</slot>
  <slot>onSpeedChanged(int)</slot>
  <slot>onStartStop()</slot>
  <slot>onTimeout()</slot>
  <slot>onRepulsionCalculationChanged(int)</slot>
  <slot>onAttractionCalculationChanged(int)</slot>
  <slot>onAccuracyChanged(int)</slot>
  <slot>onTreeToggled(bool)</slot>
  <slot>onForceApplicationChanged(int)</slot>
 </slots>
</ui>
